sklearn/utils/random.py

```diff
@@ -238,7 +238,7 @@ def random_choice_csc(n_samples, classes, class_probability=None,
         if classes[j].dtype.kind != 'i':
             raise ValueError("class dtype %s is not supported" %
                              classes[j].dtype)
-        classes[j] = classes[j].astype(int)
+        classes[j] = astype(classes[j], int, copy=False)
```

Comment:
by default this will return an array with `dtype=np.int64` what ever the `dtype` of `classes[j]`. It's better to make it explicit that we want platform independent behavior with:

```python
        classes[j] = astype(classes[j], np.int64, copy=False)
```